code/game/machinery/_machinery.dm

```diff
@@ -541,8 +541,6 @@ Class Procs:
 
 //called on machinery construction (i.e from frame to machinery) but not on initialization
 /obj/machinery/proc/on_construction()
-	for(var/obj/I in contents)
-		I.moveToNullspace()
 	return
```

Comment:
```suggestion
	if(!istype(src, /obj/machinery/computer)
		for(var/obj/I in contents)
			I.moveToNullspace()
	return
```
Maybe try this? Because it will be straight up reverting #14577 presenting even more issues.

Reply:
didnt even know that was an issue whoops
fixed it up, opening/closing doesn't drop parts and consoles work as intended